File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/PlayerListItemPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;

import io.netty.buffer.ByteBuf;

public final record PlayerListItemPacket(String playerName, boolean online, short ping) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        writeString(os, playerName);
        os.writeBoolean(online);
        os.writeShort(ping);
    }

    @Override
    public byte getId() {
        return 0x38;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/PlayerListItemPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class PlayerListItemPacket(val playerName: String, val online: Boolean, val ping: Short) : Packet.Outgoing {
    override fun serialize(os: ByteBuf) {
        os.writeString(playerName)
        os.writeBoolean(online)
        os.writeShort(ping.toInt())
    }

    override val id: Byte
        get() = 0x38
}
